_posts/2024-02-24-timekeeping

```diff
@@ -0,0 +1,108 @@
+The passage of calendar time, affecting the introduction and expiry of vehicles, house styles, and the sub-arctic snowline (if using a variable snowline NewGRF). We'll call this calendar time.
+
+Let's get the easiest out of the way first: we are leaving animation time alone.
+
+Before OpenTTD 14, economy time was not tracked separately from calendar time.
+When players wanted to run steam engines for longer, they turned on "Vehicle never expire" or used the date cheat to rewind time when needed.
+Somebody wrote a Daylength patch, later included in JGR's Patchpack, that slows down the combined calendar/economy time at the expense of many side effects.
+
+The days of such hacks are over.
+
+## The journey to real-time units
+
+In late 2022, I began the work of splitting economy and calendar time. I was following a vision from my fellow developer nielsm, who proposed a novel solution to a problem that had vexed my previous attempts at this split.
+
+The problem is this: If the economy does not follow the calendar, how do we display economy statistics like an industry's "Production last month," timetable arrival and departure dates, or the myriad other things that refer to time?
+
+The ugliest solution is to display two dates.
+But OpenTTD's user interface is confusing enough.
+Adding a second date is not a good user experience.
+
+Nielsm's proposal capitalizes on the fact that a day in OpenTTD is about two seconds, meaning that a 30-day month is about a minute.
+We can simply make this a bit more precise, then display economy time in real seconds and minutes!
+
+The first step was making the seconds-per-day a bit more precise.
+The smallest unit of time in OpenTTD is a tick, of which there are 74 in a standard day.
+Prior to OpenTTD 14, one tick lasted 27 milliseconds, making a day 2.22 seconds.
+That's close, but not quite close enough when multiplied out to a month or a year.
+But in TTD, one tick is 27 milliseconds, making a day 1.998 seconds.
+That's about as perfect as we can get.
+In OpenTTD, we've gone back to this original TTD value to better align to real-time seconds.
+
+Of course, not everyone will want to play OpenTTD using real-time units, so the new feature is an opt-in setting called Timekeeping units.
+
+Calendar units are the classic OpenTTD experience, with no changes.
+Wallclock units use minutes anywhere months are normally displayed, with 12-minute periods substituting for years. Timetables are expressed in seconds.
+
+[Image of some GUIs with wallclock unit stats]
+
+Where a date would normally be displayed (timetable start date, subsidy expiration date, etc.), the game shows an offset in seconds or minutes from the current moment.
+
+[Image of the timetable GUI showing seconds in the arrival/departure pane]
+
```

Comment:
I think you need somewhere, possibly here, a super simple tutorial about how to set the settings for different play styles. Eg. If you want time to stay the same as it always was, set time to "Calendar" mode here or here. If you want a month to last 5 minutes, then you need to set time to "Wallclock" mode and minutes per year to 12 * 5 = 60 here. If you want to stay in the same year forever, set the minutes per year to "0 (frozen time)" here. A screenshot of the settings to make this super-clear would be useful too.

Reply:
I have to agree here. There's nothing wrong with talking about the development/behind-the-scenes, but I do agree a section with some "how to use it" would be good.